lib/exrm/utils.ex

```diff
@@ -121,8 +121,32 @@ defmodule ReleaseManager.Utils do
   Get the most recent release prior to the current one
   """
   def get_last_release(project) do
-    [{_,version} | _] = project |> get_releases |> Enum.sort(fn {_, v1}, {_, v2} -> v1 > v2 end)
-    version
+    hd(project |> get_releases |> Enum.map(fn {_, v} -> v end) |> sort_versions)
+  end
+
+  @doc """
+  Sort a list of versions, latest one first. Tries to use semver version 
+  compare, but can fall back to regular string compare.
+  """
+  def sort_versions(versions) do
+    versions |> Enum.sort(
+      fn v1, v2 ->
+        case is_semver?(v1) and is_semver?(v2) do
+          true ->
+            {:ok, v1} = Version.parse(v1)
+            {:ok, v2} = Version.parse(v2)
```

Comment:
I refactored this slightly in 0d47af8. Now the parse is only performed once. We still parse and then compare, since `compare/2` will raise if the version is not valid, not that this couldn't be wrapped in a rescue, but I think this is more explicit.